addons/sale/static/src/js/product_configurator_widget.js

```diff
@@ -0,0 +1,212 @@
+odoo.define('sale.product_configurator', function (require) {
+var relationalFields = require('web.relational_fields');
+var FieldsRegistry = require('web.field_registry');
+var core = require('web.core');
+var _t = core._t;
+
+/**
+ * The sale.product_configurator widget is a simple widget extending FieldMany2One
+ * It allows the development of configuration strategies in other modules through
+ * widget extensions.
+ *
+ *
+ * !!! WARNING !!!
+ *
+ * This widget is only designed for sale_order_line creation/updates.
+ * !!! It should only be used on a product_product or product_template field !!!
+ */
+var ProductConfiguratorWidget = relationalFields.FieldMany2One.extend({
+    resetOnAnyFieldChange: true,
+    events: _.extend({}, relationalFields.FieldMany2One.prototype.events, {
+        'click .o_edit_product_configuration': '_onEditProductConfiguration'
+    }),
+
+    /**
+     * This method will check if the current product_template set on the SO line is configurable
+     * -> If so, we add a 'Edit Configuration' button next to the dropdown.
+     *
+     * @override
+     */
+     start: function () {
+        prom = this._super.apply(this, arguments);
+
+        if (this._isConfigurableProduct()) {
+            this._addConfigurationEditButton();
```

Comment:
this should be done in render i think (so that it is correctly updated when the field's value changes)